forwardproxy.go

```diff
@@ -33,294 +20,225 @@ import (
 	"sync"
 	"time"
 
-	"github.com/caddyserver/caddy/caddyhttp/httpserver"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
 	"github.com/caddyserver/forwardproxy/httpclient"
+	"golang.org/x/net/proxy"
 )
 
-type ForwardProxy struct {
-	Next httpserver.Handler
+func init() {
+	caddy.RegisterModule(Handler{})
+}
 
-	authRequired    bool
-	authCredentials [][]byte // slice with base64-encoded credentials
+type ProbeResistance struct {
+	Domain string `json:"domain,omitempty"`
+}
 
-	hideIP  bool
-	hideVia bool
+type Handler struct {
+	PACPath string `json:"pac_path,omitempty"`
 
-	pacFilePath string
+	HideIP  bool `json:"hide_ip,omitempty"`
+	HideVia bool `json:"hide_via,omitempty"`
 
-	hostname string // do not intercept requests to the hostname (except for hidden link)
-	port     string // port on which chain with forwardproxy is listening on
+	// // host and port of the proxy; i.e. when configuring a client,
+	// // you give it the host and port of the proxy to use
+	// hostname string // do not intercept requests to the hostname (except for hidden link)
+	// port     string // port on which chain with forwardproxy is listening on
+	Hosts caddyhttp.MatchHost `json:"hosts,omitempty"`
 
-	probeResistDomain  string
-	probeResistEnabled bool
+	ProbeResistance *ProbeResistance
```

Comment:
How is `probe_resistance` initialized from json config?

Reply:
Just set this field to a non-nil object.

Reply:
I can't figure out a working json config, see the errors:
`...loading module 'forward_proxy': decoding module config: http.handlers.forward_proxy: json: unknown field "probe_resistance"`

`...loading module 'forward_proxy': decoding module config: http.handlers.forward_proxy: json: unknown field "domain"`

Reply:
I think this line should add
```
ProbeResistance *ProbeResistance `json:probe_resistance,omitempty`
```
then `"probe_resistance: {"domain": "secret.localhost"}"` would work.

Reply:
Until probe resistance is finished, the key name is `ProbeResistance`.